lib/game.rb

```diff
@@ -25,45 +25,48 @@ def initialize(game_io: ConsoleIO.new, board_size:, player_2:, board_presenter:
 
     def play
         while status == :play
-            take_turn
+            output_board
+            input = get_valid_move(board: game_state.board, presenter: game_io)
+            board = update_board(input: input)
+            update_game_status(board: board)
+            game_state.switch_players if status == :play
         end
         output_board
         output_win if status == :win
         output_tie if status == :tie 
     end
 
-    def take_turn
-        output_board
-        get_valid_move
-        update_board
-        update_game_status
-        game_state.switch_players if status == :play
-    end
-
     private
+    attr_accessor :game_io, :board_presenter
+
     def output_board
-        game_io.output_board(board: game_state.board, board_presenter: board_presenter)
+        game_io.clear
+        board_presenter.generate(board: game_state.board)
+        game_io.output_message(board_presenter.output)
     end
 
-    def get_valid_move
-        @input = game_state.current_player.get_valid_move(board: game_state.board, presenter: game_io)
+    def get_valid_move(board:, presenter:)
+        input = -1
+        input = game_state.current_player.move(board: board, presenter: presenter) while !Validator.position_valid?(input: input, board: board)
+        input
     end
 
-    def update_board
+    def update_board(input:)
```

Comment:
I think it's a bit confusing that this function is called update board but then it uses the game state to update the board and then returns a board. Maybe `game_state.update` should return the updated game state?